src/library/scala/collection/StrictOptimizedSeqOps.scala

```diff
@@ -75,27 +75,38 @@ trait StrictOptimizedSeqOps [+A, +CC[_], +C]
     b.result()
   }
 
-  override def diff[B >: A](that: Seq[B]): C = {
-    val occ = occCounts(that)
-    val b = newSpecificBuilder
-    for (x <- this) {
-      val ox = occ(x)  // Avoid multiple map lookups
-      if (ox == 0) b += x
-      else occ(x) = ox - 1
+  override def diff[B >: A](that: Seq[B]): C =
+    if (that.isEmpty || this.isEmpty) fromSpecific(this)
+    else {
+      val occ = occCounts(that)
+      val b = newSpecificBuilder
+      for (x <- this) {
+        occ.updateWith(x) {
+          case None => {
+            b.addOne(x)
+            None
+          }
+          case Some(1) => None
+          case Some(n) => Some(n - 1)
+        }
+      }
+      b.result()
     }
-    b.result()
-  }
 
-  override def intersect[B >: A](that: Seq[B]): C = {
-    val occ = occCounts(that)
-    val b = newSpecificBuilder
-    for (x <- this) {
-      val ox = occ(x)  // Avoid multiple map lookups
-      if (ox > 0) {
-        b += x
-        occ(x) = ox - 1
+  override def intersect[B >: A](that: Seq[B]): C =
+    if (that.isEmpty || this.isEmpty) fromSpecific(this)
```

Comment:
shouldn't this be `empty` rather than `fromSpecific(this)`? if `this` isn't empty, then it's not the intersection

```suggestion
    if (that.isEmpty || this.isEmpty) empty
```

Reply:
Yes, I think that's the cause of the failing partest run. I'll include some more junit tests